emission/analysis/classification/inference/mode.py

```diff
@@ -48,15 +50,80 @@ def runPipeline(self):
     (self.featureMatrix, self.resultVector) = self.generateFeatureMatrixAndResultVectorStep()
     logging.info("generateFeatureMatrixAndResultVectorStep DONE")
     (self.cleanedFeatureMatrix, self.cleanedResultVector) = self.cleanDataStep()
+   
     logging.info("cleanDataStep DONE")
     self.selFeatureIndices = self.selectFeatureIndicesStep()
     logging.info("selectFeatureIndicesStep DONE")
     self.selFeatureMatrix = self.cleanedFeatureMatrix[:,self.selFeatureIndices]
     self.model = self.buildModelStep()
     logging.info("buildModelStep DONE")
-    toPredictTripsQuery = {"$and": [{'type': 'move'},
-                                    ModeInferencePipeline.getModeQuery(''),
-                                    {'predicted_mode': None}]}
+    # toPredictTripsQuery = {"$and": [{'type': 'move'},
+    #                                ModeInferencePipeline.getModeQuery(''),
+     #                               {'predicted_mode': None}]]}
+    Algorithm_to_be_used_to_predict = 0
+
+    #If we want to predict the things who haven't been predicted with the current prediction method, we can
+    #   set that here. If it is zero, that means predict everything that doesn't have a prediction already in the database
+    if Algorithm_to_be_used_to_predict:
+      IDsWithThisAlgorithm = PredictionDB.find().distinct("section_id", { 'algorithm_id': Algorithm_to_be_used_to_predict})
+      toPredictIDsQuery = {'_id': {'$nin' : IDsWithThisAlgorithm}}
+
+    else:
+      AllIDsInPredictionDB = PredictionDB.find().distinct("section_id")
+      toPredictIDsQuery = {'_id' : {'$nin': AllIDsInPredictionDB}}
+
+    #toPredictTripsQuery = {"$and": [ModeInferencePipeline.getModeQuery(''), {'$or': ['predicted_mode': None}]}
+    
+    #This returns all the section ids that are not currently in the predictiondb
+    #We're going to need to either query the section database to see which section does not have a prediction in the
+    #   prediction database, or we need to grab all the ids of the predictions whose algorithm_id is zero.
+
+    (self.toPredictFeatureMatrix, self.sectionIds, self.sectionUserIds) = self.generateFeatureMatrixAndIDsStep(toPredictTripsQuery)
+    logging.info("generateFeatureMatrixAndIDsStep DONE")
+    self.predictedProb = self.predictModesStep()
+    #This is a matrix of the entries and their corresponding probabilities for each classification
+
+    logging.info("predictModesStep DONE")
+    self.savePredictionsStep()
+    logging.info("savePredictionsStep DONE")
+
+
+  def runPipeline(self):
```

Comment:
I created `runPipelineModelStage` because running the `runPipeline` in `model_stage` wouldn't have done what we needed, so I made a new version of the same function where I could play with the function without deleting the version used in `testPipeline`.